### PR TITLE
WIP initial mobile client view on overview screen

### PR DESCRIPTION
**Pull request by maleck13**
@sedroche here are my changes so far

### CODE REVIEW

**Comment by maleck13** [comment]
**Trying it out**

set up the minishift server
target the minishift server with this branch
create a mobile client 
It should show up on the overview
expand the client you should see an option for browsing the catalog
delete the mobile client (should alert and remove the client)

File: app/scripts/app.js [hunk 1]
```diff
@@ -634,6 +634,24 @@ angular
...
+          { id: 'apps', label: 'Apps', tags: ['mobile'], icon: 'fa fa-mobile' },
+         {
+           id: 'services',
+          label: 'Services',
+          tags: ['mobile-service'],
+          icon: 'fa fa-database'
+        }]
+      });
+    }
     Logger.info("AEROGEAR_MOBILE_ENABLED: " + $rootScope.AEROGEAR_MOBILE_ENABLED);
```
Comment (witmicko) [inline app/scripts/app.js]: I feel that this block might grow in the future, we probably should move it to separate file and call it from here if possible.

File: app/scripts/directives/overview/mobileClientRow.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      apiObject: '<',
+      state: '<',
+      bindings: '<'
+    },
+    templateUrl: 'views/overview/_mobile-client-row.html'
+  });
+
+  function MobileAppRow($scope, $filter, $routeParams, ProjectsService, DataService, APIService, MobileClientsService, ListRowUtils, Navigate, AuthorizationService) {
+    var row = this;
+    $scope.installType = '';
```
Comment (sedroche) [inline app/scripts/directives/overview/mobileClientRow.js]: Think we should pick one of `$scope` or `controllerAs` in a component and stick to it.

File: app/scripts/directives/overview/mobileClientRow.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    _.extend(row, ListRowUtils.ui);
+    
+    row.bundleDisplay = "bundle id: " + row.apiObject.spec.appIdentifier;
+    switch (row.apiObject.spec.clientType) {
+      case 'android':
+        row.icon = 'fa fa-android';
+        $scope.installType = 'maven';
```
Comment (sedroche) [inline app/scripts/directives/overview/mobileClientRow.js]: Does `class="{{row.apiObject.metadata.labels.icon}}"` on the relevant element work?
If it doesn't creating a filter for this makes sense

File: app/scripts/services/mobileClients.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        gracePeriodSeconds: 0
+      }).then(
+        // success
+        _.noop,
+        // failure
+        function (err) {
+          NotificationsService.addNotification({
+            id: "remove-mobileclient-error",
+            type: "error",
+            message: "An error occurred while deleting the mobile client" + apiObject.metadata.name + ".",
```
Comment (sedroche) [inline app/scripts/services/mobileClients.js]: Should we wrap these property references in `_.get`?

File: app/scripts/directives/overview/mobileClientRow.js (hunk 2 above)
Comment (maleck13) [reply app/scripts/directives/overview/mobileClientRow.js]: sure

**Pull request merged**